### PR TITLE
feat(auto-salvage): v2 support

### PR DESCRIPTION
**Pull request by c3y1huang**
#### Which issue(s) this PR fixes:

Issue longhorn/longhorn#8430

#### What this PR does / why we need it:

Include snapshot disk info in the proxy.ReplicaList response.

#### Special notes for your reviewer:

`None`

#### Additional documentation or context

`None`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,6 +2,12 @@ module github.com/longhorn/longhorn-instance-manager
 
 go 1.22.2
 
+replace github.com/longhorn/longhorn-spdk-engine => github.com/c3y1huang/longhorn-spdk-engine v0.0.0-20240927005842-eccd2551cd2e
+
+replace github.com/longhorn/types => github.com/c3y1huang/types v0.0.0-20240927004343-3ad221f0cbfc
+
+replace github.com/longhorn/go-common-libs => github.com/c3y1huang/go-common-libs v0.0.0-20240927004544-ee33d7bb917c
```
Comment (c3y1huang) [inline go.mod:9]: TODO: update when dependency PRs are merged.

File: go.mod [hunk 2]
```diff
@@ -2,14 +2,16 @@ module github.com/longhorn/longhorn-instance-manager
 
 go 1.22.2
 
+replace github.com/longhorn/longhorn-spdk-engine => github.com/c3y1huang/longhorn-spdk-engine v0.0.0-20241101010916-59483ca8824a
```
Comment (c3y1huang) [inline go.mod:5]: TODO: update when dependency PRs are merged.

**Review by derekbit — APPROVED** [review]
LGTM. Waiting for the merge of longhorn-spdk-engine's PR.

**Review by derekbit — APPROVED** [review]
LGTM. Waiting for the merge of longhorn-spdk-engine's PR.

**Review by derekbit — APPROVED** [review]

**Pull request merged**